src/main/java/week6/RearrangingCars.java

```diff
@@ -0,0 +1,120 @@
+package week6;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+public class RearrangingCars {
+
+  private static final int EMPTY = 0;
+
+  /*
+   * First, we create a map of wrongly positioned cars, car no. -> position
+   * Iteratively, replace 0 with the correct car. We remove cars that are in correct pos from map
+   *
+   * If 0 becomes in the correct position but map is not empty, move 0 to first incorrect car pos
+   * and start iterating again until map is empty.
+   *
+   * Time Complexity: O(n)
+   * Space Complexity: O(n)
+   * Assumptions:
+   *    1) len(source) = len(target)
+   *    2) arrays contain O...n, this means no duplications
+   *
+   * where n is the length of the array
+   */
+  public static List<Integer> pathForEmpty(final int[] source, final int[] target) {
+
+    if (source == null || target == null || source.length != target.length) {
+      return new ArrayList<>();
+    }
+
+    //creates map (wrongly positioned) car id -> position and get index of the empty Box
+    HashMap<Integer, Integer> cars = new HashMap<>();
```

Comment:
I've seen several places with //, there is some standard saying use /* */
I agree it is silly, but that's one of the comments I got when Simon reviewed my code... I guess it will be also the case here...